beacon_node/eth2_libp2p/src/discovery/mod.rs

```diff
@@ -519,69 +549,116 @@ impl<TSpec: EthSpec> Discovery<TSpec> {
-            .map(|v| QueryResult(query, v));
+            .map(|v| QueryResult(queries, v));
 
         // Add the future to active queries, to be executed.
         self.active_queries.push(Box::pin(query_future));
     }
 
     /// Drives the queries returning any results from completed queries.
-    fn poll_queries(&mut self, cx: &mut Context) -> Option<(Option<Instant>, Vec<Enr>)> {
+    fn poll_queries(&mut self, cx: &mut Context) -> Option<HashMap<PeerId, Option<Instant>>> {
+        let mut results: Vec<(Option<Instant>, Vec<Enr>)> = Vec::new();
+
         while let Poll::Ready(Some(query_future)) = self.active_queries.poll_next_unpin(cx) {
-            match query_future.0 {
-                QueryType::FindPeers => {
-                    self.find_peer_active = false;
-                    match query_future.1 {
-                        Ok(r) if r.is_empty() => {
-                            debug!(self.log, "Discovery query yielded no results.");
-                        }
-                        Ok(r) => {
-                            debug!(self.log, "Discovery query completed"; "peers_found" => r.len());
-                            return Some((None, r));
+            // Iterate through queries comprising the original grouped query
+            query_future.0.clone().into_iter().for_each(|query| {
+                let query_future_enrs = query_future.1.clone();
+                match query {
+                    QueryType::FindPeers => {
+                        self.find_peer_active = false;
+                        match query_future_enrs {
+                            Ok(r) if r.is_empty() => {
+                                debug!(self.log, "Discovery query yielded no results.");
+                            }
+                            Ok(r) => {
+                                debug!(self.log, "Discovery query completed"; "peers_found" => r.len());
+                                results.push((None, r));
+                            }
+                            Err(e) => {
+                                warn!(self.log, "Discovery query failed"; "error" => e.to_string());
+                            }
                         }
-                        Err(e) => {
-                            warn!(self.log, "Discovery query failed"; "error" => e.to_string());
+                    }
+                    QueryType::Subnet(subnet_query) => {
+                        match query_future_enrs {
+                            Ok(r) if r.is_empty() => {
+                                debug!(self.log, "Subnet discovery query yielded no results."; "subnet_id" => *subnet_query.subnet_id, "retries" => subnet_query.retries);
+                            }
+                            Ok(r) => {
+                                let enr: Vec<Enr> = r.clone();
+                                debug!(self.log, "Peer subnet discovery request completed"; "peers_found" => r.len(), "subnet_id" => *subnet_query.subnet_id);
```

Comment:
Should move this log until after the filter. Otherwise we will get invalid logs for each subnet query we requested